class Solution {
    public long sumAndMultiply(int n) {
        if(n==0) return 0;
        String str= String.valueOf(n);
        StringBuilder sb= new StringBuilder("");
        int sum=0;
        for(int i=0;i<str.length();i++){
            int ch=Integer.valueOf(str.charAt(i)-'0');
            if(ch!=0){
                sb.append(ch);
                sum+=ch;
            }
        }
        long x= Long.valueOf(sb.toString())*sum;
        return x;
    }
}
